common/transport/amqp/src/amqp.ts

```diff
@@ -48,64 +51,34 @@ export class Amqp {
         False if the caller intends to manually settle messages
         A string containing the version of the SDK used for telemetry purposes] */
   constructor(autoSettleMessages: boolean) {
-    const autoSettleMode = autoSettleMessages ? amqp10.Constants.receiverSettleMode.autoSettle : amqp10.Constants.receiverSettleMode.settleOnDisposition;
+    const autoSettleMode = autoSettleMessages ? 1 : 2;
     // node-amqp10 has an automatic reconnection/link re-attach feature that is enabled by default.
     // In our case we want to control the reconnection flow ourselves, so we need to disable it.
 
     /*Codes_SRS_NODE_COMMON_AMQP_16_042: [The Amqp constructor shall create a new `amqp10.Client` instance and configure it to:
     - not reconnect on failure
     - not reattach sender and receiver links on failure
     - not reestablish sessions on failure]*/
-    this._amqp = new amqp10.Client(amqp10.Policy.merge(<any>{
-      session: {
-        reestablish: {
-          retries: 0,
-          forever: false
-        }
-      },
-      senderLink: {
-        attach: {
-          maxMessageSize: 0,
-        },
-        encoder: (body: any): any => {
-          if (typeof body === 'string') {
-            return new Buffer(body, 'utf8');
-          } else {
-            return body;
-          }
-        },
-        reattach: {
-          retries: 0,
-          forever: false
-        }
-      },
-      receiverLink: {
-        attach: {
-          maxMessageSize: 0,
-          receiverSettleMode: autoSettleMode,
-        },
-        decoder: (body: any): any => body,
-        reattach: {
-          retries: 0,
-          forever: false
-        }
-      },
-      // reconnections will be handled at the client level, not the transport level.
-      reconnect: {
-        retries: 0,
-        strategy: 'fibonacci',
-        forever: false
-      }
-    }, amqp10.Policy.EventHub));
-
+    this._amqpContainer = rhea.create_container(
+    );
+    let manageConnectedConnectionHandlers: (operation: string) => void;
     const amqpErrorHandler = (err) => {
-      debug('amqp10 client error: ' + err.toString());
+      debug('rhea client error: ' + ((!!err) ? err.toString() : 'no err provided on the error'));
       this._fsm.handle('amqpError', err);
     };
+    const amqpDisconnectedHandler = (err) => {
+      debug('rhea client error: ' + ((!!err) ? err.toString() : 'no err provided on the disconnect'));
+      this._fsm.handle('amqpDisconnected', err);
+    };
+    manageConnectedConnectionHandlers = (operation: string) => {
+      this._amqpConnection[operation]('error', amqpErrorHandler);
+      this._amqpConnection[operation]('disconnected', amqpDisconnectedHandler);
+      this._amqpConnection[operation]('connection_close', amqpErrorHandler);
```

Comment:
>amqpErrorHandler [](start = 58, length = 16)

that feels wrong. `connection_close` should not necessarily be an error situation